lib/weather/presentation/weather_details.dart

```diff
@@ -1,10 +1,361 @@
+                    ),
+                  ],
+                ),
+                const SizedBox(
+                  height: 12,
+                ),
+                const DayDataItem(
+                  dayOfWeek: 'Monday',
+                  imageUrl: 'assets/images/big_rain_drops.png',
+                  maxTemp: 13,
+                  minTemp: 10,
+                ),
+                const DayDataItem(
+                  dayOfWeek: 'Tuesday',
+                  imageUrl: 'assets/images/cloud_3_zap.png',
+                  maxTemp: 17,
+                  minTemp: 12,
+                ),
+                const DayDataItem(
+                  dayOfWeek: 'Wednesday',
+                  imageUrl: 'assets/images/cloud_3_zap.png',
+                  maxTemp: 21,
+                  minTemp: 19,
+                ),
+              ],
+            ),
+          ],
+        ),
+      ),
+    );
+  }
+}
+
+class DayDataItem extends StatelessWidget {
+  const DayDataItem({
+    super.key,
+    required this.dayOfWeek,
+    required this.imageUrl,
+    required this.maxTemp,
+    required this.minTemp,
+  });
+
+  final String dayOfWeek;
+  final String imageUrl;
+  final int maxTemp;
+  final int minTemp;
+
+  @override
+  Widget build(BuildContext context) {
+    return SizedBox(
+      height: 50,
+      child: Row(
+        children: [
+          Text(
+            dayOfWeek,
+            textAlign: TextAlign.start,
+            style: GoogleFonts.alegreyaSans(
+              textStyle: const TextStyle(
+                fontSize: 18,
+                height: 21.6 / 18,
+                fontWeight: FontWeight.w700,
+                color: blackColor,
+              ),
+            ),
+          ),
+          Expanded(
+            child: Row(
+              mainAxisAlignment: MainAxisAlignment.end,
+              children: <Widget>[
+                Padding(
+                  padding: const EdgeInsets.only(right: 87),
+                  child: Image.asset(
+                    imageUrl,
+                    alignment: Alignment.center,
+                  ),
+                ),
+                TemperatureData(
+                  color: blackColor,
+                  data: maxTemp,
+                  dataFontSize: 18,
+                  dataLineHeight: 21.6,
+                  unitFontSize: 10,
+                  unitLineHeight: 12,
+                ),
+                const SizedBox(
+                  width: 10,
+                ),
+                TemperatureData(
+                  color: blackColor.withOpacity(0.5),
+                  data: minTemp,
+                  dataFontSize: 18,
+                  dataLineHeight: 21.6,
+                  unitFontSize: 10,
+                  unitLineHeight: 12,
+                ),
+              ],
+            ),
+          )
+        ],
+      ),
+    );
+  }
+}
+
+class ForecastByHourItem extends StatelessWidget {
+  const ForecastByHourItem({
+    super.key,
+    required this.data,
```

Comment:
data has too general meaning. It should be named more precisely

Reply:
poor naming for things is what I try to mitigate, thanks for pointing it out.